src/Collectors/JobCollector.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace AG\ElasticApmLaravel\Collectors;
+
+use AG\ElasticApmLaravel\Contracts\DataCollector;
+use Illuminate\Queue\Events\JobFailed;
+use Illuminate\Queue\Events\JobProcessed;
+use Illuminate\Queue\Events\JobProcessing;
+use Illuminate\Support\Facades\Log;
+use PhilKra\Events\Transaction;
+use Throwable;
+
+/**
+ * Collects info about the job process.
+ */
+class JobCollector extends TimelineDataCollector implements DataCollector
+{
+    public function getName(): string
+    {
+        return 'job-collector';
+    }
+
+    protected function registerEventListeners(): void
+    {
+        $this->app->events->listen(JobProcessing::class, function (JobProcessing $event) {
+            $transaction_name = $this->getTransactionName($event);
+            $this->startTransaction($transaction_name);
+            $this->setTransactionType($transaction_name);
+        });
+
+        $this->app->events->listen(JobProcessed::class, function (JobProcessed $event) {
+            $transaction_name = $this->getTransactionName($event);
+            $this->stopTransaction($transaction_name);
+            $this->setTransactionResult($transaction_name, 200);
+        });
+
+        $this->app->events->listen(JobFailed::class, function (JobFailed $event) {
+            $transaction_name = $this->getTransactionName($event);
+            $this->stopTransaction($transaction_name);
+            $this->setTransactionResult($transaction_name, 400);
```

Comment:
If a job failed is usually because of an exception right? If that's the case, I would rather prefer a 500 status code to make it really visible. 4XX codes can regular app validation response. @countless-integers what do you think?

Reply:
Same. I'd keep 4XX errors for client generated errors only and in this case it's our app's wiring that causes the issue.